src/types/pairlists.ts

```diff
@@ -0,0 +1,55 @@
+import { BackgroundTaskResult } from './backgroundtasks';
+import { WhitelistResponse } from './blacklist';
+import { MarginMode, TradingMode } from './types';
+
+export interface PairlistsResponse {
+  pairlists: Pairlist[];
+}
+
+export interface PairlistEvalResponse extends BackgroundTaskResult {
+  result: WhitelistResponse;
+}
+
+export interface Pairlist {
+  id?: string;
+  is_pairlist_generator: boolean;
+  name: string;
+  description: string;
+  params: Record<string, PairlistParameter>;
+}
+
+export interface PairlistConfig {
+  name: string;
+  blacklist: string[];
+  pairlists: Pairlist[];
+}
+
+export enum PairlistParamType {
+  string = 'string',
+  number = 'number',
+  boolean = 'boolean',
+  option = 'option',
+}
+
+export interface PairlistParameter {
+  description: string;
+  help: string;
+  type: PairlistParamType;
+  value?: string;
+  default: string;
```

Comment:
Types for this are wrong - the value / default of a parameter can be `string | number | boolean | string[]` - depending on the type (which means we should probably split this into 3-4 dedicated types to ensure type safety).

As a consequence, This also makes the type of the v-model for PairlistConfigParameter wrong, which is defined as string - not as `string | boolean | number`.
i suspect that this implicitly causes the "staticPairlistDefault" bug (which shows the parameter as checked, while in reality it isn't) - as we're feeding an empty string into it, instead of true/false.

Reply:
Fixed ... :laughing: